runtime/src/committee.rs

```diff
@@ -0,0 +1,752 @@
+
+impl<T: Trait<I>, I: Instance> Module<T, I> {
+    /// Returns true if given did is contained in `Members` set. `false`, otherwise.
+    pub fn is_member(who: &IdentityId) -> bool {
+        Self::members().contains(who)
+    }
+
+    /// Given `votes` number of votes out of `total` votes, this function compares`votes`/`total`
+    /// in relation to the threshold proporion `n`/`d`.
+    fn is_threshold_satisfied(
+        votes: u32,
+        total: u32,
+        (threshold, n, d): (ProportionMatch, u32, u32),
+    ) -> bool {
+        match threshold {
+            ProportionMatch::AtLeast => votes * d >= n * total,
+            ProportionMatch::MoreThan => votes * d > n * total,
+        }
+    }
+}
+
+pub struct EnsureProportionMoreThan<N: U32, D: U32, AccountId, I = DefaultInstance>(
+    rstd::marker::PhantomData<(N, D, AccountId, I)>,
+);
+impl<
+        O: Into<Result<RawOrigin<AccountId, I>, O>> + From<RawOrigin<AccountId, I>>,
+        N: U32,
+        D: U32,
+        AccountId,
+        I,
+    > EnsureOrigin<O> for EnsureProportionMoreThan<N, D, AccountId, I>
+{
+    type Success = ();
+    fn try_origin(o: O) -> Result<Self::Success, O> {
+        o.into().and_then(|o| match o {
+            RawOrigin::Members(n, m) if n * D::VALUE > N::VALUE * m => Ok(()),
+            r => Err(O::from(r)),
+        })
+    }
+}
+
+pub struct EnsureProportionAtLeast<N: U32, D: U32, AccountId, I = DefaultInstance>(
+    rstd::marker::PhantomData<(N, D, AccountId, I)>,
+);
+impl<
+        O: Into<Result<RawOrigin<AccountId, I>, O>> + From<RawOrigin<AccountId, I>>,
+        N: U32,
+        D: U32,
+        AccountId,
+        I,
+    > EnsureOrigin<O> for EnsureProportionAtLeast<N, D, AccountId, I>
+{
+    type Success = ();
+    fn try_origin(o: O) -> Result<Self::Success, O> {
+        o.into().and_then(|o| match o {
+            RawOrigin::Members(n, m) if n * D::VALUE >= N::VALUE * m => Ok(()),
+            r => Err(O::from(r)),
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{balances, committee, identity};
+    use core::result::Result as StdResult;
+    use hex_literal::hex;
+    use sr_io::with_externalities;
+    use sr_primitives::{
+        testing::Header,
+        traits::{BlakeTwo256, Block as BlockT, ConvertInto, IdentityLookup, Verify},
+        AnySignature, BuildStorage, Perbill,
+    };
+    use srml_support::{
+        assert_noop, assert_ok,
+        dispatch::{DispatchError, DispatchResult},
+        parameter_types, Hashable,
+    };
+    use substrate_primitives::{Blake2Hasher, H256};
+    use system::EnsureSignedBy;
+    use system::{self, EventRecord, Phase};
+    use test_client::{self, AccountKeyring};
+
+    parameter_types! {
+        pub const BlockHashCount: u64 = 250;
+        pub const MaximumBlockWeight: u32 = 1024;
+        pub const MaximumBlockLength: u32 = 2 * 1024;
+        pub const AvailableBlockRatio: Perbill = Perbill::one();
+    }
+    impl system::Trait for Test {
```

Comment:
Instead of define a new `Test` in each module, we could use `TestStorage` (in `test/storage.rs`).
You could also use your own `make_ext()` function to create the externalities or update the function inside `test/storage.rs` if you think it would be useful/needed for tests in other modules.

See `test/identity.rs` as an example of `TestStorage`.

Reply:
@fmiguelgarcia Since we are testing using an instance of the module, we have a new runtime defined in the tests. `Test` is defined with `construct_runtime` and other traits are implemented on this `Test`. I believe TestStorage isn't setup for instantiable modules, is it?